adal/src/androidTest/java/com/microsoft/aad/adal/AcquireTokenRequestTest.java

```diff
@@ -558,6 +599,433 @@ public void testVerifyBrokerRedirectUriInvalidSignature() throws PackageManager.
+
+        try {
+            assertNull(callback.getCallbackResult());
+            assertNotNull(callback.getCallbackException());
+            assertTrue(callback.getCallbackException() instanceof AuthenticationException);
+            assertTrue(((AuthenticationException) callback.getCallbackException()).getCode() == ADALError.AUTH_FAILED_NO_TOKEN);
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for Resiliency Feature when the AT is not expired and outage mode is off
+     *
+     * If
+     * 1. outage mode is off, which means ExtendedLifetimeEnabled is set to be false in AuthenticationContext
+     * 2. the token cache item is not expired
+     * 3. the server is good
+     * Then
+     * An valid authentication result will be returned.
+     */
+    @SmallTest
+    public void testResiliencyTokenReturn_outageModeOff_validAT() throws PackageManager.NameNotFoundException,
+            NoSuchAlgorithmException, OperationCanceledException, IOException, AuthenticatorException,
+            InterruptedException {
+        // make sure AT's expires_in is expired and ext_expires_in is not expired
+        final Calendar expiredTime = new GregorianCalendar();
+        expiredTime.add(Calendar.MINUTE, MINUS_MINUITE);
+        final ITokenCacheStore cacheStore = getTokenCache(expiredTime.getTime(), false, false);
+
+        final AccountManager mockedAccountManager = getMockedAccountManager();
+        mockAccountManagerGetAccountBehavior(mockedAccountManager);
+        final FileMockContext mockContext = createMockContext();
+        mockContext.setMockedAccountManager(mockedAccountManager);
+
+        final AuthenticationContext authContext = new AuthenticationContext(mockContext,
+                VALID_AUTHORITY, false, cacheStore);
+        //turn on the outage mode
+        authContext.setExtendedLifetimeEnabled(false);
+        //mock HttpUrlConnection for refresh token request
+        prepareSuccessHttpUrlConnection();
+        //set useBroker into false
+        AuthenticationSettings.INSTANCE.setUseBroker(false);
+
+        final TestAuthCallback callback = new TestAuthCallback();
+        authContext.acquireTokenSilentAsync("resource", "clientid", TEST_USERID, callback);
+
+        final CountDownLatch signal = new CountDownLatch(1);
+        signal.await(ACTIVITY_TIME_OUT, TimeUnit.MILLISECONDS);
+
+        try {
+            assertNull(callback.getCallbackException());
+            assertNotNull(callback.getCallbackResult());
+            assertTrue(callback.getCallbackResult().getAccessToken().equals("I am an AT"));
+        } finally {
+            cacheStore.removeAll();
+        }
+    }
+
+    /**
+     * Test for the retry when encountering with 500/503/504
+     * The request should be posted twice with 500/503/504 error from server
+     */
+     public void testExtendedLifetimeRequest_retry_positive() throws PackageManager.NameNotFoundException,
```

Comment:
Sorry, but i don't understand the difference of this test and others. Almost all of your tests have retry, and what is the difference here? Verify the number time that getInputStream should be verified in each retry tests, we shouldn't create so many duplication here.